CourseSketchProjects/CourseSketchDatabaseServer/src/main/java/handlers/subhandlers/GradingRequestHandler.java

```diff
@@ -74,15 +74,21 @@ private GradingRequestHandler() {
         LOG.debug("Query State instructor: {}, student: {}, allGrades: {}, singleGrade: {}", instructor, student, allGrades, singleGrade);
         List<ProtoGrade> returnList = new ArrayList<>();
         if (instructor && allGrades) {
-            returnList = institution.getAllAssignmentGradesInstructor(request.getItemId(COURSE_INDEX), userId);
+            returnList = institution.getAllAssignmentGradesInstructor(request.getItemId(COURSE_INDEX), authId);
         } else if (student && allGrades) {
-            returnList = institution.getAllAssignmentGradesStudent(request.getItemId(COURSE_INDEX), userId);
-        } else if (instructor && singleGrade) {
-            returnList.add(institution.getGrade(userId, request.getItemId(USER_INDEX), request.getItemId(COURSE_INDEX),
-                    request.getItemId(ASSIGNMENT_INDEX), request.getItemId(PROBLEM_INDEX)));
-        } else if (student && singleGrade) {
-            returnList.add(institution.getGrade(userId, userId, request.getItemId(COURSE_INDEX), request.getItemId(ASSIGNMENT_INDEX),
-                    request.getItemId(PROBLEM_INDEX)));
+            returnList = institution.getAllAssignmentGradesStudent(request.getItemId(COURSE_INDEX), authId);
+        } else {
+            final ProtoGrade gradeData = ProtoGrade.newBuilder()
+                    .setUserId(request.getItemId(USER_INDEX))
+                    .setCourseId(request.getItemId(COURSE_INDEX))
+                    .setAssignmentId(request.getItemId(ASSIGNMENT_INDEX))
+                    .setProblemId(request.getItemId(PROBLEM_INDEX))
+                    .build();
+            if (instructor && singleGrade) {
+                returnList.add(institution.getGrade(authId, gradeData));
+            } else if (student && singleGrade) {
```

Comment:
Okay after I looked into it they actually do do the same thing and there was a bug in the grade manager code that is fixed now!